plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -530,7 +530,7 @@ private void indexRemoteModel(
                     .build();
 
                 IndexRequest indexModelMetaRequest = new IndexRequest(ML_MODEL_INDEX);
-                if (registerModelInput.getIsHidden()) {
+                if (registerModelInput.getIsHidden() != null && registerModelInput.getIsHidden())  {
```

Comment:
Check if any other places using `registerModelInput.getIsHidden()` and add null checking?

Reply:
Found another one, added.